### PR TITLE
IN-5 Implement repository pattern

### PR DESCRIPTION
**Pull request by glachezar**
Let me know if you need me to add something else, as I couldn't think of anything else to add.

### CODE REVIEW

File: investment-app/TRINV.Api/TRINV.Infrastructure/Configurations/DependencyInjection.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using Interfaces;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Repositories;
+
+public static class DependencyInjection
+{
+    public static IServiceCollection AddInfrastructure(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddScoped<IApplicationDbContextFactory, ApplicationDbContextFactory>();
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Infrastructure/Configurations/DependencyInjection.cs:13]: Do we need this ?

File: investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private readonly ApplicationDbContext _context;
+
+    public Repository(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<T>> GetAllAsync()
+    {
+        return await _context.Set<T>().ToArrayAsync();
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs:18]: Better change it to ListAsync().
It easy to work with IList collections

File: investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public async Task<T?> GetByIdAsync(int id)
+    {
+        return await _context.Set<T>().FindAsync(id);
+    }
+
+    public void Insert(T entity)
+    {
+        _context.Set<T>().Add(entity);
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs:28]: Maybe will be useful to return the id of the inserted record

File: investment-app/TRINV.Api/TRINV.Application/Interfaces/IUnitOfWork.cs [hunk 4]
```diff
@@ -2,4 +2,5 @@
 
 public interface IUnitOfWork
 {
+    Task SaveChangesAsync(CancellationToken cancellationToken = default);
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Application/Interfaces/IUnitOfWork.cs:5]: You sure this unit of work will work just like that ?

File: investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs (hunk 2 above)
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs:18]: Done.

**Pull request merged**